v2/cpu.go

```diff
@@ -16,108 +16,23 @@
 
 package v2
 
-import (
-	"bufio"
-	"io/ioutil"
-	"os"
-	"path/filepath"
-	"strconv"
-
-	statsv2 "github.com/containerd/cgroups/v2/stats"
-	specs "github.com/opencontainers/runtime-spec/specs-go"
-)
-
-func NewCpu(unifiedMountpoint string) (*cpuController, error) {
-	c := &cpuController{
-		unifiedMountpoint: unifiedMountpoint,
-	}
-
-	ok, err := c.Available("/")
-	if err != nil {
-		return nil, err
-	}
-	if !ok {
-		return nil, ErrCPUNotSupported
-	}
-	return c, nil
-}
-
-type cpuController struct {
-	unifiedMountpoint string
-}
-
-func (c *cpuController) Name() Name {
-	return Cpu
-}
-
-func (c *cpuController) path(g GroupPath) string {
-	return filepath.Join(c.unifiedMountpoint, string(g))
+type CPU struct {
+	Weight *uint64
+	Max    *uint64
 }
 
-func (c *cpuController) Create(g GroupPath, resources *specs.LinuxResources) error {
-	if err := os.MkdirAll(c.path(g), defaultDirPerm); err != nil {
-		return err
+func (r *CPU) Values() (o []CGStr) {
+	if r.Weight != nil {
+		o = append(o, CGStr{
+			key:   "cpu.weight",
+			value: *r.Weight,
+		})
 	}
-	if cpuShares := resources.CPU.Shares; cpuShares != nil {
-		// Converting cgroups configuration from v1 to v2
-		// more here https://github.com/containers/crun/blob/master/crun.1.md#cgroup-v2
-		convertedWeight := (1 + ((*cpuShares-2)*9999)/262142)
```

Comment:
where is the new code for this?

Reply:
i'm skipping this for now to make sure we have a simple API first, then we can port the spec -> resource type code over along with stats.  The old API is too complicated and has too much code.